storybookjs/storybook
# Toolbars: Add dynamicTitle option

Issue: #17790 

## What I did

- Add a `dynamicTitle` option to display the title of the selected item in the Toolbar.
- This feature is opt in so it does not disrupt anyone's existing UI

![toolbar-dynamic-title](https://user-images.githubusercontent.com/1775017/159825503-c0b746b7-cd98-4b67-b063-cf56a4bc7393.gif)

## How to test

1. Add `dynamicTitle: true` to any global type
2. Change the value in the toolbar
3. The Toolbar should display the title of the selected item

Example with a `theme` globalType
```js
export const globalTypes = {
  theme: {
    name: 'Theme',
    description: 'Global theme for components',
    toolbar: {
      dynamicTitle: true,
      icon: 'circlehollow',
      title: 'Theme',
      items: [
        { value: 'light', icon: 'circlehollow', title: 'Light' },
        { value: 'dark', icon: 'circle', title: 'Dark' },
      ],
    },
  },
};
```
